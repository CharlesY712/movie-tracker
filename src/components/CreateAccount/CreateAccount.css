.account-page {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-flow: row wrap;
  background-color: var(--card-bg);
  color: var(--font-color);
  padding: 10px;
  border-radius: 10px;
  border: 1px solid var(--card-border);
  margin: 5px auto;
  min-height: 100px;
  min-width: 145px;
  width: 50%;
  margin: 3em auto;
}

.account-page input {
  margin: 10px;
  height: 3em;
  width: 16em;
}

.button {
  width: 85px;
  height: 3.5em;
}