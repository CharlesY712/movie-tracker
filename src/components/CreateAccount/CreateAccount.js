import React, { Component } from 'react';
import { postCreateUser, signIn } from '../../helper/apiCall';
import PropTypes from 'prop-types';

class CreateAccount extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: ''
    };
  }

  handleChange = (event) => {
    const { name, value } = event.target;

    this.setState({
      [name]: value
    });
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const createUser = await postCreateUser(this.state);
    createUser.status ?
      this.handleNewUser(event):
      this.handleError(createUser.error);

  }

  handleNewUser = async (event) => {
    const { email, password } = this.state;
    const grabUser = await signIn(event, email, password);
    this.props.handleUser(grabUser.data);
    this.props.history.push('/');
  }

  handleError = () => {
    alert('Email already exists');
  }

  render() {
    const { name, email, password } = this.state;
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <input type="text"
            name="name"
            value={name}
            placeholder="Enter Your name"
            onChange={this.handleChange}
            required
          />
          <input
            type="email"
            name="email"
            value={email}
            placeholder="Enter Email"
            onChange={this.handleChange}
            required
          />
          <input
            type="password"
            name="password"
            value={password}
            placeholder="Enter Your Password"
            onChange={this.handleChange}
            required
          />
          <button type="submit">Submit</button>
        </form>
      </div>
    );
  }
}
CreateAccount.propTypes = {

  handleUser: PropTypes.func,
  history: PropTypes.object
};

export default CreateAccount;
