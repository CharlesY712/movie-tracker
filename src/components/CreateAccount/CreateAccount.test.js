import React from 'react';
import { shallow } from 'enzyme';
import CreateAccount from './CreateAccount';


describe('CreateAccount', () => {
  let wrapper, mockHandleUser, mockPush;

  beforeEach(() => {
    mockHandleUser = jest.fn();
    mockPush = jest.fn();
    wrapper = shallow(
      <CreateAccount
        handleUser={mockHandleUser}
        history={ {push:mockPush} } />)
  })

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('Should have a default state of name as a empty string', () => {
    expect(wrapper.state('name')).toEqual('')
  })

  it('Should have a default state of email as a empty string', () => {
    expect(wrapper.state('email')).toEqual('')
  })

  it('Should have a default state of password as a empty string', () => {
    expect(wrapper.state('password')).toEqual('')
  })

  it('should set state of the event value when a change occurs', () => {
    const mockEvent = { target: {
      name:'name',
      value:'Bob'
    }};
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('name')).toEqual('Bob');
  })

  it('Should call handleNewUSer if there is not an error', async () => {

    const user = {
      status: "success",
      data: {
        id: 13,
        name: "bob",
        password: "bugs",
        email: "jhonson@aol.col"
      }
    };

    const mockEvent = { preventDefault: jest.fn()};

    // const mockGrabUser = jest.fn().mockImplementation(() => Promise.resolve ({user}))

    // {status: "success", data: {â€¦}, message: "Retrieved ONE User"}
    // data: {
    //   {id: 13, name: "bob", password: "bugs", email: "jhonson@aol.col"}

    // }

    const mockHandleNewUser = jest.fn()

    wrapper.instance().handleSubmit(mockEvent)

    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        ok: true,
        json: () => Promise.resolve(results)
    }));

    
    const spy = jest.spyOn(wrapper.instance(), 'handleNewUser')

    expect(spy).toHaveBeenCalledWith(mockEvent)

  })

  
})

