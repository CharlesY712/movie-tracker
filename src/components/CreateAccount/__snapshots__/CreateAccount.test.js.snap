// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateAccount
    handleUser={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
>
        <input
                name="name"
                onChange={[Function]}
                placeholder="Enter Your name"
                required={true}
                type="text"
                value=""
        />
        <input
                name="email"
                onChange={[Function]}
                placeholder="Enter Email"
                required={true}
                type="email"
                value=""
        />
        <input
                name="password"
                onChange={[Function]}
                placeholder="Enter Your Password"
                required={true}
                type="password"
                value=""
        />
        <button
                type="submit"
        >
                Submit
        </button>
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="name"
            onChange={[Function]}
            placeholder="Enter Your name"
            required={true}
            type="text"
            value=""
/>,
          <input
            name="email"
            onChange={[Function]}
            placeholder="Enter Email"
            required={true}
            type="email"
            value=""
/>,
          <input
            name="password"
            onChange={[Function]}
            placeholder="Enter Your Password"
            required={true}
            type="password"
            value=""
/>,
          <button
            type="submit"
>
            Submit
</button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "name",
            "onChange": [Function],
            "placeholder": "Enter Your name",
            "required": true,
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "Enter Email",
            "required": true,
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "Enter Your Password",
            "required": true,
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
>
          <input
                    name="name"
                    onChange={[Function]}
                    placeholder="Enter Your name"
                    required={true}
                    type="text"
                    value=""
          />
          <input
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter Email"
                    required={true}
                    type="email"
                    value=""
          />
          <input
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter Your Password"
                    required={true}
                    type="password"
                    value=""
          />
          <button
                    type="submit"
          >
                    Submit
          </button>
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="name"
              onChange={[Function]}
              placeholder="Enter Your name"
              required={true}
              type="text"
              value=""
/>,
            <input
              name="email"
              onChange={[Function]}
              placeholder="Enter Email"
              required={true}
              type="email"
              value=""
/>,
            <input
              name="password"
              onChange={[Function]}
              placeholder="Enter Your Password"
              required={true}
              type="password"
              value=""
/>,
            <button
              type="submit"
>
              Submit
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "onChange": [Function],
              "placeholder": "Enter Your name",
              "required": true,
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Enter Email",
              "required": true,
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Enter Your Password",
              "required": true,
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
